cartographer_grpc/data_uploader.cc

```diff
@@ -16,8 +16,46 @@
 
 #include "cartographer_grpc/data_uploader.h"
 
+#include "glog/logging.h"
+
 namespace cartographer_grpc {
 
-DataUploader::DataUploader(const std::string &data_uploader) {}
+DataUploader::DataUploader(const std::string& server_address)
+    : client_channel_(grpc::CreateChannel(server_address,
+                                          grpc::InsecureChannelCredentials())),
+      service_stub_(proto::MapBuilderService::NewStub(client_channel_)) {}
+
+void DataUploader::AddTrajectory(
+    int local_trajectory_id,
+    const std::unordered_set<std::string>& expected_sensor_ids,
+    const cartographer::mapping::proto::TrajectoryBuilderOptions&
+        trajectory_options) {
+  grpc::ClientContext client_context;
+  proto::AddTrajectoryRequest request;
+  proto::AddTrajectoryResponse result;
+  *request.mutable_trajectory_builder_options() = trajectory_options;
+  for (const auto& sensor_id : expected_sensor_ids) {
+    *request.add_expected_sensor_ids() = sensor_id;
+  }
+  grpc::Status status =
+      service_stub_->AddTrajectory(&client_context, request, &result);
+  CHECK(status.ok());
+  CHECK_EQ(local_to_cloud_trajectory_id_map_.count(local_trajectory_id), 0);
```

Comment:
Can it be non-zero? We have so many CHECK macros in the code-base. All they all necessary?

Reply:
It would be a mistake by the caller to call this function with a `trajectory_id` that has been previously added.